src/main.rs

```diff
@@ -1,52 +1,94 @@
 extern crate docopt;
 extern crate rand;
 extern crate rustc_serialize;
-#[macro_use] extern crate lazy_static;
 
-use std::process;
+use std::{process, io, path};
+use std::fs::File;
+use std::io::Read;
 use rand::distributions::{IndependentSample, Range};
 use docopt::Docopt;
 
 const USAGE: &'static str = "
 Generate XKCD style passwords.
 
 Usage:
-  xkcdpass [-c <count>]
+  xkcdpass [options] [-w <list>...]
 
 Options:
-  -h --help     Show this screen.
-  -v --version  Show version.
-  -c <count>    Number of words [default: 4].
+  -h --help                 Show this screen.
+  -v --version              Show version.
+  -c <count>                Number of words [default: 4].
+  -w --use-wordlists <list>
 ";
 
-static WORDLIST: &'static str = include_str!("wordlist.txt");
-lazy_static! {
-    static ref MAX_OFFSET: usize = WORDLIST.trim().chars().filter(|c| *c as u32 == 0x0a).count();
-    static ref BETWEEN: Range<usize> = Range::new(0, *MAX_OFFSET);
-}
+static DEFAULT_WORDLIST: &'static str = include_str!("wordlist.txt");
 
 #[derive(Debug, RustcDecodable)]
 struct Args {
     flag_c: usize,
+    flag_w: Option<Vec<String>>,
+}
+
+fn load_wordlist(path: String) -> io::Result<String> {
+    let path = path::Path::new(&path);
+    let mut file = File::open(path)?;
+    let mut string = String::new();
+    file.read_to_string(&mut string)?;
+    Ok(string)
+}
+
+fn load_wordlists(paths: Vec<String>) -> io::Result<String> {
+    let mut size = 0;
+    let wordlists = paths.into_iter()
+        .map(|p| {
+            let list = load_wordlist(p);
+            if let Ok(ref s) = list {
+                size += s.len();
+            }
+            list
+        })
+        .collect::<Result<Vec<_>, _>>()?;
+
+    let mut string = String::with_capacity(size);
+
+    for list in wordlists {
+        string.push_str(&list);
+    }
+
+    Ok(string)
 }
 
-fn get_random_word() -> &'static str {
+fn get_random_word(list: &str, offset: usize) -> &str {
     let mut rng = rand::os::OsRng::new().unwrap_or_else(|e| {
         println!("Could not initialize random number generator: {}", e);
         process::exit(1);
     });
-    let offset = BETWEEN.ind_sample(&mut rng);
-    WORDLIST.lines().nth(offset).expect(&format!("Invalid offset: {}", offset))
+    let between = Range::new(0, offset);
+    let offset = between.ind_sample(&mut rng);
+    list.lines().nth(offset).expect(&format!("Invalid offset: {}", offset))
 }
 
 fn main() {
     let args: Args = Docopt::new(USAGE)
-                            .and_then(|d| d.decode())
-                            .unwrap_or_else(|e| e.exit());
+        .and_then(|d| d.decode())
+        .unwrap_or_else(|e| e.exit());
+
+    let list = if let Some(lists) = args.flag_w {
+        match load_wordlists(lists) {
+            Ok(l) => l,
+            Err(e) => {
+                println!("{}", e);
+                process::exit(1);
+            }
```

Comment:
You can use `unwrap_or_else` for this:

```rust
load_wordlists(lists).unwrap_or_else(|e| {
    println!("{}", e);
    process::exit(1);
})
```

The first `if let` could also be converted to a `.map`, but that probably makes it harder to read.